docs/aws/audit/redshiftmonitoring/rules/redshift_parameter_group_require_ssl_remediation.mdx

```diff
@@ -1,6 +1,207 @@
+
+2. **Access Parameter Groups:**
+   - In the Amazon Redshift console, on the left-hand navigation pane, click on "Parameter groups."
+
+3. **Modify Parameter Group:**
+   - Select the parameter group you want to modify.
+   - Click on the "Edit" button to modify the parameters of the selected parameter group.
+
+4. **Set require_ssl Parameter:**
+   - In the list of parameters, find the parameter named `require_ssl`.
+   - Change its value to `true`.
+   - Save the changes by clicking the "Save changes" button.
+
+By following these steps, you ensure that SSL is required for connections to your Amazon Redshift cluster, enhancing the security of your data in transit.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration of Redshift Parameter Group not requiring SSL in AWS using the AWS CLI, follow these steps:
+
+1. **Create a New Parameter Group:**
+   First, create a new Redshift parameter group if you don't already have one. This will allow you to customize the parameters, including requiring SSL.
+
+   ```sh
+   aws redshift create-cluster-parameter-group \
+       --parameter-group-name my-redshift-parameter-group \
+       --parameter-group-family redshift-1.0 \
+       --description "My custom Redshift parameter group"
+   ```
+
+2. **Modify the Parameter Group to Require SSL:**
+   Update the parameter group to set the `require_ssl` parameter to `true`.
+
+   ```sh
+   aws redshift modify-cluster-parameter-group \
+       --parameter-group-name my-redshift-parameter-group \
+       --parameters "ParameterName=require_ssl,ParameterValue=true,ApplyType=static"
+   ```
+
+3. **Apply the Parameter Group to Your Redshift Cluster:**
+   Apply the modified parameter group to your Redshift cluster. This will enforce the SSL requirement.
+
+   ```sh
+   aws redshift modify-cluster \
+       --cluster-identifier my-redshift-cluster \
+       --cluster-parameter-group-name my-redshift-parameter-group
+   ```
+
+4. **Reboot the Redshift Cluster:**
+   Reboot the Redshift cluster to apply the new parameter settings.
+
+   ```sh
+   aws redshift reboot-cluster \
+       --cluster-identifier my-redshift-cluster
+   ```
+
+By following these steps, you ensure that your Redshift cluster requires SSL for connections, thereby preventing the misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of Redshift Parameter Group not requiring SSL in AWS Redshift using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK for Python (Boto3):**
+   - Ensure you have Boto3 installed. If not, install it using pip:
+     ```bash
+     pip install boto3
+     ```
+
+2. **Create a Boto3 Session:**
+   - Initialize a session with your AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>